### PR TITLE
Add models for, and merge in, raw configs provided by ACS

### PR DESCRIPTION
**Pull request by euank**
Since this isn't actually something ACS does right now, this has not yet been tested beyond the unit tests.

If we'd like to hold off on merging this or merge it into a branch other than `dev`, I'd understand. Just would like to review sooner rather than later.

r? @samuelkarp

### CODE REVIEW

File: agent/api/task.go [hunk 1]
```diff
@@ -220,13 +220,28 @@ func (task *Task) dockerConfig(container *Container) (*docker.Config, *DockerCli
...
+		err := json.Unmarshal([]byte(*container.DockerConfig.Config), &config)
+		if err != nil {
+			return nil, &DockerClientConfigError{"Unable decode given docker config: " + err.Error()}
+		}
+	}
+
+	// Augment labels with some metadata from the agent. Explicitly do this last
+	// such that it will always override duplicates in the provided raw config
+	// data.
+	for key, value := range map[string]string{
```
Comment (samuelkarp) [inline agent/api/task.go]: Can you just change this to

```
config.Labels["com.amazonaws.ecs.task-arn"] = task.Arn
config.Labels["com.amazonaws.ecs.container-name"] = container.Name
config.Labels["com.amazonaws.ecs.task-definition-family"] = task.Family
config.Labels["com.amazonaws.ecs.task-definition-version"] = task.Version
```
?

I don't think the loop is particularly readable or adds significant value.

File: agent/api/task_test.go [hunk 2]
```diff
@@ -194,6 +198,83 @@ func TestDockerHostConfigVolumesFrom(t *testing.T) {
 	}
 }
 
+func TestDockerHostConfigRawConfig(t *testing.T) {
```
Comment (samuelkarp) [inline agent/api/task_test.go]: Can you add a test similar to this showing that non-overridden existing data in Config/HostConfig is properly preserved when merged?

**Comment by euank** [comment]
Addressed your comments @samuelkarp

**Comment by samuelkarp** [comment]
:+1:

**Pull request merged**